### PR TITLE
TASK Refactor ContentContext methods

### LINKED ISSUE #31: `ContentContext::getCurrentRenderingMode` to `RenderingModeService::findByCurrentUser`
```diff
- $renderingMode = $context->getCurrentRenderingMode();
+ $renderingMode = $this->renderingModeService->findByCurrentUser();
```

### PR DESCRIPTION
**Pull request by dlubitz**
This rector replaces removed ContentContext calls in PHP and Fusion. For better detection of context objects all Node(Interface)s get replaced by `Neos\Rector\ContentRepository90\Legacy\NodeLegacyStub`. The `Neos\Rector\ContentRepository90\Legacy\NodeLegacyStub` get replaced at the end by `Neos\ContentRepository\Core\Projection\ContentGraph\Node`

PHP
* `ContentContext::isInBackend` => `$this->renderingModeService->findByCurrentUser()->isEdit`
* `ContentContext::isLive` => `!$this->renderingModeService->findByCurrentUser()->isEdit && !$this->renderingModeService->findByCurrentUser()->isPreview`
* `ContentContext::getCurrentRenderingMode` => `$this->renderingModeService->findByCurrentUser()`
* `ContentContext::getCurrentSiteNode` => Warning, no replacement
* `ContentContext::getCurrentDomain` => Warning, no replacement
* `ContentContext::getCurrentSite` => Warning, no replacement

Fusion
* `node.context.currentDomain` => Warning, no replacement
* `context.currentSiteNode` => Warning, no replacement

Fixes #31

### CODE REVIEW

File: config/set/contentrepository-90.php [hunk 1]
```diff
@@ -275,23 +279,22 @@
      * ContentContext
      */
     // ContentContext::getCurrentSite
-    // TODO: PHP
+    $methodCallToWarningComments[] = new MethodCallToWarningComment(LegacyContextStub::class, 'getCurrentSite', '!! ContentContext::getCurrentSite() is removed in Neos 9.0.');
```
Comment (dlubitz) [inline config/set/contentrepository-90.php:282]: What can we suggest here as replacement?

File: config/set/contentrepository-90.php [hunk 2]
```diff
@@ -275,23 +279,22 @@
...
     $rectorConfig->rule(FusionContextCurrentSiteRector::class);
-    // TODO: Fusion
     // ContentContext::getCurrentDomain
-    // TODO: PHP
-    // TODO: Fusion
+    $methodCallToWarningComments[] = new MethodCallToWarningComment(LegacyContextStub::class, 'getCurrentDomain', '!! ContentContext::getCurrentDomain() is removed in Neos 9.0.');
```
Comment (dlubitz) [inline config/set/contentrepository-90.php:285]: What can we suggest here as replacement?

File: config/set/contentrepository-90.php [hunk 3]
```diff
@@ -275,23 +279,22 @@
...
+    $fusionNodePropertyPathToWarningComments[] = new FusionNodePropertyPathToWarningComment('context.currentDomain', 'Line %LINE: !! node.context.currentDomain is removed in Neos 9.0.');
```
Comment (dlubitz) [inline config/set/contentrepository-90.php:286]: What can we suggest here as replacement?

File: config/set/contentrepository-90.php [hunk 4]
```diff
@@ -275,23 +279,22 @@
...
     // ContentContext::getCurrentSiteNode
-    // TODO: PHP
-    // TODO: Fusion
+    $methodCallToWarningComments[] = new MethodCallToWarningComment(LegacyContextStub::class, 'getCurrentSiteNode', '!! ContentContext::getCurrentSiteNode() is removed in Neos 9.0.');
```
Comment (dlubitz) [inline config/set/contentrepository-90.php:288]: What can we suggest here as replacement?

File: config/set/contentrepository-90.php [hunk 5]
```diff
@@ -275,23 +279,22 @@
...
+    $fusionNodePropertyPathToWarningComments[] = new FusionNodePropertyPathToWarningComment('context.currentSiteNode', 'Line %LINE: !! node.context.currentSiteNode is removed in Neos 9.0.');
```
Comment (dlubitz) [inline config/set/contentrepository-90.php:289]: What can we suggest here as replacement?

File: config/set/contentrepository-90.php (hunk 4 above)
Comment (kitsunet) [reply config/set/contentrepository-90.php:288]: Grab subgraph and traverse up to site node, at least as hint how to do it

**Review by kitsunet — APPROVED** [review]
definitely an improvement.

**Review by ahaeslich — APPROVED** [review]

**Pull request merged**